# WES Genotyping Pipeline Purely in Bash (WES-GENPIPIB)

This repository contains a workflow used in-house for discovering single nucleotide variants (SNVs) from whole exome sequencing (WES) data.

## Table of Contents
- [Introduction](#introduction)
- [Installation](#installation)
- [Usage](#usage)
- [Workflow Overview](#workflow-overview)
- [Contributing](#contributing)
- [License](#license)

## Introduction
This workflow is designed to identify SNVs from WES data generated in-house. It leverages various bioinformatics tools packaged in GATK Suite and follows GATK Best Practices. 

## Installation
Activate the `gatk` Mamba environment already installed in this AMI before running the Bash pipeline.

## Prerequisites

1. Create a data manifest tab-separated text file containing file prefix, run ID, and sample ID to be used with the Bash pipeline. Below is an example of one.

	```
	FILE	TUM_ID	SLX_ID
	14499.i704_i503	SD0507	14499
	14336.i710_i504	SD0560	14336
	15093.i704_i502	SD0683	15093
	14499.i702_i507	SD0693	14499
	14336.i709_i505	SD1043	14336
	```
   **NOTE**: The pipeline operates under the assumption that raw data are stored in the cloud in AWS S3, so the manifest file is needed (and the only input needed) for the script to work. The first line of the file can be a header; the script WILL ignore the first line so a headerless manifest will lead to data skipping. 

2. The `refs` directory contains all the required reference files to run the pipeline. Due to github limitation, they are not uploaded in this repo, but their exact filenames can be found below:

	```
	GRCh38-109_bwa_db.amb
	GRCh38-109_bwa_db.ann
	GRCh38-109_bwa_db.bwt
	GRCh38-109_bwa_db.pac
	GRCh38-109_bwa_db.sa
	GRCh38.109.fa
	GRCh38.109.fa.fai
	GRCh38.109.dict
	Homo_sapiens_assembly38.dbsnp138.renamed.vcf
	Homo_sapiens_assembly38.known_indels.renamed.vcf
	Mills_and_1000G_gold_standard.indels.hg38.renamed.vcf

	```
3. The `...bwa_db` index files are generated by BWA by running the command below on the main `ref.fa` file (`GRCh38.109.fa`) in this case. Only takes less than 30 minutes to finish.

	> bwa index -p [db.prefix] ref.fa 

4. The `GRCh38.109.fa` reference file needs to be indexed for GATK tools to use. Run the command below prior to running the Bash pipeline. This will create a `ref.fa.fai` index file.

	> samtools faidx ref.fa -@ [INT - number of cores to use] 

5. The GATK tools also require a dictionary file (`GRCh38.109.dict`) to be generated from the `ref.fa` file. Run this command to generate it prior to running the Bash pipeline.

	> gatk CreateSequenceDictionary -R ref.fa

6. Finally, the three `vcf` files are annotation files that would be used during BQSR application and haplotype calling steps. 

	These VCF files need to be indexed as well using the bundled tool in the GATK tool suite, `IndexFeatureFile`. Run the commands below prior to running the pipeline. 

	> gatk IndexFeatureFile -I refs/Mills_and_1000G_gold_standard.indels.hg38.renamed.vcf

	> gatk IndexFeatureFile -I refs/Homo_sapiens_assembly38.dbsnp138.renamed.vcf

	> gatk IndexFeatureFile -I refs/Homo_sapiens_assembly38.known_indels.renamed.vcf

Sources: 
1. [GATK FASTA Reference Genome Format](https://gatk.broadinstitute.org/hc/en-us/articles/360035531652-FASTA-Reference-genome-format)

2. [GATK IndexFeatureFile](https://gatk.broadinstitute.org/hc/en-us/articles/360037428111-IndexFeatureFile)

## Usage
To execute the workflow, just execute the main script in the repo after cloning the whole repository off Github. 

Using `mamba run` persists the mamba environments into subshells so the internal commands in the script can find the installed tools.

```
mamba run -n gatk ./WES-GENPIPIB.sh -o /absolute/path/to/output/dir -d -j 4 /home/ubuntu/repos/WES-SNV-discovery-workflow/manifests/WES-TUM-preprocessing-manifest.txt
```
`-d` here stands for `--dry-run` while the option `-j` specifies the number of parallel jobs you would like GNU Parallel to run for each stage of the pipeline. 

Other options can be grokked at by running `mamba run -n gatk ./WES-GENPIPIB.sh -h`, which would return a help message.


## Workflow Overview
The workflow consists of the following steps:
1. **Read Trimming**: Trim the raw FASTQ reads with Trim Galore
2. **Read Alignment**: Align trimmed reads to the reference genome using BWA-MEM
3. **Read Group ID Addition**: Add or edit RG fields in the resulting BAMs
4. **Sample ID/Tumor IDâ€“specific Bam Filename Collation**: Collate and save BAM files corresponding to a particular Sample ID or Tumor ID into a text file
5. **Bam File Merging**: Merge these BAM files at Tumor ID level using GATK bundled tool
6. **Duplicate Read Marking**: Mark duplicate reads in the merged BAM files using a GATK bundled tool
7. **N CIGAR Read Splitting**: Split N CIGAR reads
8. **Base Quality Score Recalibration (BQSR) Modelling**: Model BQSR for recalibration of quality scores
9. **BSQR Application**: Apply the modeled BSQR to the BAM files
10. **Germline Haplotype Calling**: Call germline haplotypes on the aligned BAM files

## Contributing
Contributions are welcome! Please submit a pull request or open an issue to discuss your ideas.

## License
This project is licensed under the MIT License. See the [LICENSE](LICENSE) file for details.
