# WES SNV Discovery Workflow

This repository contains a workflow used in-house for discovering single nucleotide variants (SNVs) from whole exome sequencing (WES) data.

## Table of Contents
- [Introduction](#introduction)
- [Installation](#installation)
- [Usage](#usage)
- [Workflow Overview](#workflow-overview)
- [Contributing](#contributing)
- [License](#license)

## Introduction
This workflow is designed to identify SNVs from WES data generated in-house. It leverages various bioinformatics tools packaged in GATK Suite and follows GATK Best Practices. 

## Installation
Activate the `gatk` Mamba environment in this AMI before running the Bash workflow.

## Prerequisites

1. Create a data manifest tab-separated file containing file prefix, run ID, and sample ID to be used with the Bash pipeline. **NOTE**: The pipeline operates under the assumption that raw data are stored in the cloud in AWS S3, so the manifest file is needed (and the only input needed) for the script to work. The first line of the file can be a header; the script WILL ignore the first line so a headerless manifest will lead to data skipping. 

2. The `refs` directory contains all the required reference files to run the pipeline. Due to github limitation, they are not uploaded in this repo, but their exact filenames can be found below:

```
GRCh38-109_bwa_db.amb
GRCh38-109_bwa_db.ann
GRCh38-109_bwa_db.bwt
GRCh38-109_bwa_db.pac
GRCh38-109_bwa_db.sa
GRCh38.109.fa
GRCh38.109.fa.fai
GRCh38.109.dict
Homo_sapiens_assembly38.dbsnp138.renamed.vcf
Homo_sapiens_assembly38.known_indels.renamed.vcf
Mills_and_1000G_gold_standard.indels.hg38.renamed.vcf

```
* The `...bwa_db` index files are generated by BWA by running the command below on the main `ref.fa` file (`GRCh38.109.fa`) in this case. Only takes less than 30 minutes to finish.

> bwa index -p [db.prefix] ref.fa 

* The `GRCh38.109.fa` reference file needs to be indexed for GATK tools to use. Run the command below prior to running the Bash pipeline. This will create a `ref.fa.fai` index file.

> samtools faidx ref.fa -@ [INT - number of cores to use] 

* The GATK tools also require a dictionary file (`GRCh38.109.dict`) to be generated from the `ref.fa` file. Run this command to generate it prior to running the Bash pipeline.

> gatk CreateSequenceDictionary -R ref.fa

* Finally, the three `vcf` files are annotation files that would be used during BQSR application and haplotype calling steps. They can be downloaded elsewhere on the Internet.

## Usage
To execute the workflow, just run the following Bash script.
```
bash WES-DNA-SNV-CRMY.sh
```
NOTE: Make sure that `bwa index [ref.fa]` has been run first to generate the BWA-compatible reference genome indices. 


## Workflow Overview
The workflow consists of the following steps:
1. **Pre-Processing**: Assess the quality of raw sequencing data.
2. **Alignment**: Align reads to the reference genome.
3. **Variant Calling**: Identify SNVs from the aligned reads.
4. **Annotation**: Annotate the identified variants.

## Contributing
Contributions are welcome! Please submit a pull request or open an issue to discuss your ideas.

## License
This project is licensed under the MIT License. See the [LICENSE](LICENSE) file for details.